.github/workflows/thread-sanitizer.yaml

```
@@ -1,142 +1,148 @@
 name: Thread Sanitizer analysis
 on:
   workflow_dispatch:
+    inputs:
+      fastdds_branch:
+        description: 'Branch or tag of Fast DDS repository (https://github.com/eProsima/Fast-DDS)'
+        required: true
+        default: 'master'
+      optional_cmake_args:
+        description: 'Optional CMake Compilation Flags'
+        required: false
+        type: string
+      optional_ctest_args:
+        description: 'Optional CTest Testing Flags'
+        required: false
+        type: string
+
   push:
     branches:
-      - 'feature/tsan/fixes'
+      - 'master'
+    paths-ignore:
+      - '**.md'
+      - '**.txt'
+      - '!**/CMakeLists.txt'
+
   pull_request:
     branches:
-      - 'feature/tsan/fixes'
-env:
-  TSAN_OPTIONS: second_deadlock_stack=1
-  # ubuntu host doesn't have an environment SHELL value (docker and vm images does)???
-  SHELL: /bin/bash
-
-defaults:
-  run:
-    shell: pwsh
-    working-directory: /home/runner/work
+      - 'master'
```

If we want Thread Sanitizer to run on PRs to 2.6.x, this should be changed
```suggestion
      - '2.6.x'
```

---

bee5f0c39